src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -351,53 +352,40 @@ def _deploy_arm_template_core_unmodified(cmd, resource_group_name, template_file
     deployment_client = smc.deployments  # This solves the multi-api for you
 
     if not template_uri:
-
         # pylint: disable=protected-access
         deployment_client._serialize = JSONSerializer(
             deployment_client._serialize.dependencies
         )
 
         # Plug this as default HTTP pipeline
-        from msrest.pipeline import Pipeline
-        from msrest.pipeline.requests import (
-            RequestsCredentialsPolicy,
-            RequestsPatchSession,
-            PipelineRequestsHTTPSender
-        )
-        from msrest.universal_http.requests import RequestsHTTPSender
-
-        smc.config.pipeline = Pipeline(
-            policies=[
-                JsonCTemplatePolicy(),
-                smc.config.user_agent_policy,
-                RequestsPatchSession(),
-                smc.config.http_logger_policy,
-                RequestsCredentialsPolicy(smc.config.credentials)
-            ],
-            sender=PipelineRequestsHTTPSender(RequestsHTTPSender(smc.config))
+        from azure.core.pipeline import Pipeline
+        smc._client._pipeline._impl_policies.append(JsonCTemplatePolicy())
```

Comment:
Consider using `per_call_policies` introduced by https://github.com/Azure/azure-sdk-for-python/pull/17340.